chainstate/src/detail/tx_verification_strategy.rs

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) 2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use chainstate_types::{BlockIndex, BlockIndexHandle};
+use common::{
+    chain::{Block, ChainConfig, GenBlock},
+    primitives::{id::WithId, Amount, BlockHeight, Id, Idable},
+};
+use tx_verifier::transaction_verifier::{
+    error::ConnectTransactionError, storage::TransactionVerifierStorageRef, BlockTransactableRef,
+    Fee, Subsidy, TransactionVerifier,
+};
+use utils::tap_error_log::LogError;
+
+use crate::{calculate_median_time_past, BlockError};
+
+pub struct TransactionVerificationStrategy<'a, H, S> {
+    block_index_handle: &'a H,
+    storage_backend: &'a S,
+    chain_config: &'a ChainConfig,
+}
+
+impl<'a, H: BlockIndexHandle, S: TransactionVerifierStorageRef>
+    TransactionVerificationStrategy<'a, H, S>
+{
+    pub fn new(
+        block_index_handle: &'a H,
+        storage_backend: &'a S,
+        chain_config: &'a ChainConfig,
+    ) -> Self {
+        Self {
+            block_index_handle,
+            storage_backend,
+            chain_config,
+        }
+    }
+
+    pub fn connect_block(
```

Comment:
maybe there should be `trait TransactionVerificationStartegy` so that we can implement different strategies